src/coreclr/nativeaot/Runtime/arm64/AsmMacros.h

```diff
@@ -81,6 +81,10 @@ TrapThreadsFlags_TrapThreads     equ 2
 TrapThreadsFlags_AbortInProgress_Bit equ 0
 TrapThreadsFlags_TrapThreads_Bit     equ 1
 
+;; Bit position for the ARM64IntrinsicConstants_Atomics flags, to be used with tbz / tbnz instructions
+;; ARM64IntrinsicConstants_Atomics = 0x0080
+ARM64_ATOMICS_FEATURE_FLAG_BIT       equ 7
```

Comment:
Can we validate that the value of this constant matches the C++ version?

This looks very likely to get forgotten if we ever shuffle the enum for whatever reason.

We have a mechanism for compile-time validation of these things in AsmOffsets.h. It's not an offset per se, but this is used to ensure `MAX_STRING_LENGTH` agree, etc.

Reply:
I'll put this into offsets (with appropriate check).

Reply:
ARM64_ATOMICS_FEATURE_FLAG_BIT is in the AsmOffsets.h and there is an assert that couples it with ARM64IntrinsicConstants_Atomics